integration-tests/application-server-integration-tests/src/test/resources/catalina.policy

```diff
@@ -0,0 +1,274 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
```

Comment:
Is this tomcat policy file the same for all Tomcat versions ? I would be quite surprised if it was the case, and it might defeat the purpose of testing with "the tomcat server as the users will experience it". Also, it would make it harder to do the same with other containers.

If possible, I would prefer if we could just append the last few lines to an existing policy file, maybe creating a policy on the fly in `enableSecurityManager`.

```
grant codeBase "file:/elastic-apm-agent.jar" {
    permission java.security.AllPermission;
};
```

Reply:
> Is this tomcat policy file the same for all Tomcat versions ? 

Not 100%, there are some minor small additions of specific permissions through versions. Does it matter? 

> I would be quite surprised if it was the case, and it might defeat the purpose of testing with "the tomcat server as the users will experience it". 

How would it defeat the purpose? The purpose is to see whether the agent functions as expected without causing issues when the security manager is enabled. The permissions within the policy are only allowing basic Tomcat stuff and web apps to function properly.
The way I know it does not defeat the purpose is that it helped me identify those places where additional changes in the agent code were still missing. The tests fail without these fixes.

> Also, it would make it harder to do the same with other containers.

This one is only adding to Tomcat and it already proved very efficient as it helped identifying additional required fixes. It is not a base for adding to other Servlet containers anyway as it relies on `JAVA_OPTS`, which is specific to Tomcat's startup scripts.

> If possible, I would prefer if we could just append the last few lines to an existing policy file, maybe creating a policy on the fly in enableSecurityManager.

The effective policy at runtime is a superset of all used policies. To my understanding, this includes policies added through `-Djava.security.policy=...` and using `-Djava.security.policy==...` would override other policies. I also wanted to do the same, only adding a policy that grants permissions to our agent, assuming like you that the Tomcat image includes a default policy file, but apparently it doesn't, as this didn't work without providing a full policy.